src/Multipart/ResponseBuilder.php

```diff
@@ -51,25 +52,32 @@ public function __invoke(
         $mainPart = null;
         $attachments = $attachmentStorage->responseAttachments();
         foreach ($document->getParts() as $part) {
+            // When no "start" is provided, the first part should be considered the main part.
+            // @see https://datatracker.ietf.org/doc/html/rfc2387#section-3.2
+            if (null === $mainPart && null === $start) {
+                $mainPart = $part;
+                continue;
+            }
+
             $mimeType = $part->getMimeType();
-            $id = string()->coerce($part->getHeader('Content-ID'));
+            $id = string()->coerce($part->getHeader('Content-ID', ''));
 
-            if (($start && $id === $start) || $mimeType === $soapType) {
```

Comment:
I did add some kind of fallback here for when start was not set, but I don't think it was properly tested before.
The reason why this wasn't working is because it has main type of xop with a subtype of text/xml and only the main type was validated against text/xml.

However, i think it makes more sense to drop this check and use the first part instead.

The XOP spec states that the start-info MUST match, so I suppose there is no need to actually validate this or look up alternate parts. Because an attachment might as well be XML. (see https://www.w3.org/TR/xop10/#mime_xop_packages)